airgun/views/common.py

```diff
@@ -1,8 +1,48 @@
-from widgetastic.widget import View
+import six
+from widgetastic.widget import NoSuchElementException, View, WidgetMetaclass
 
-from ..widgets import HorizontalNavigation, ContextSelector
+from airgun.widgets import ContextSelector, HorizontalNavigation, Search
 
 
 class BaseLoggedInView(View):
     menu = HorizontalNavigation()
     taxonomies = ContextSelector()
+
+
+class WidgetMixin(six.with_metaclass(WidgetMetaclass, object)):
+    """Base class for all View and Widget mixins"""
+    # todo: remove it and use widgetastic native one, once it's introduced
+    pass
+
+
+class SearchableViewMixin(WidgetMixin):
+    """Mixin which adds :class:`airgun.widgets.Search` widget and
+    :meth:`search` to your view. It's useful for _most_ entities list views
+    where searchbox is present.
+
+    Note that you can override expected result locator for the element which is
+    returned by :meth:`search` by specifying custom ``search_result_locator``
+    string variable in your view class.
+    """
+    searchbox = Search()
+    search_result_locator = "//a[contains(., '%s')]"
```

Comment:
It is overridable, right?

Reply:
Take a closer look at docstring ;)
Yes, it's overridable, that's one of main points of such mixin.

Reply:
right